### DISCUSSION TITLE
VRoidアバターVRMモデルうまくロードできない現象について

### DISCUSSION THREAD

**Discussion by wcl1207** [pixiv/vroid-sdk-developers#115] [Q&A] [answered]
Exampleシーンで Onicco_suika_v1.0.3アバターが下図のようにうまくロードできないようですが、原因がわかりそうでしょうか。
<img width="1136" alt="スクリーンショット 2023-10-25 15 55 07" src="https://github.com/pixiv/vroid-sdk-developers/assets/32382938/4320ba7d-7c82-48fa-bf93-89d44f5e23e2">

気になったのは、PickUp一覧にOnicco_suikaのようなアバターが他にも存在する場合、何かの対処を考えなければならないです。

Exampleシーン:Single PlayerのVRoidSDK/Examples/CharacterModelExample/Scenes/Example
Onicco_suika_v1.0.3アバター：https://hub.vroid.com/characters/7592926366680666240/models/2494976733431136106

**Comment by nyamadan** [comment]
> Onicco_suika_v1.0.3アバター：[https://hub.vroid.com/characters/7592926366680666240/models/2494976733431136106](https://hub.vroid.com/characters/7592926366680666240/models/2494976733431136106)

こちらのモデルですが手元では事象が再現しませんでした。モデルの読み込みはできていて、レンダリングのフェーズに失敗しているように見えますが…
何かコンソールにエラーなどは出ていますでしょうか？

**Comment by wcl1207** [comment]
早速のご調査、ありがとうございます！

コンソールにエラーなどが出ていないですが、materialの読み込みは失敗のようです。

ちなみにですが、バージョン情報は下記の通りです。
UniVRM：[0.111](https://github.com/vrm-c/UniVRM/releases/)
VRoid SDK：0.3.0
VRoidマルチプレイSDK：0.3.0
Unity：2020.3.43f1


<img width="757" alt="スクリーンショット 2023-10-26 9 24 00" src="https://github.com/pixiv/vroid-sdk-developers/assets/32382938/1e21d992-53dc-4201-a59b-92d6a1873259">

<img width="757" alt="スクリーンショット 2023-10-26 9 23 29" src="https://github.com/pixiv/vroid-sdk-developers/assets/32382938/bd0f7d36-270f-444e-9d05-401596bfdf03">

<img width="1509" alt="スクリーンショット 2023-10-26 9 23 44" src="https://github.com/pixiv/vroid-sdk-developers/assets/32382938/5de3ca73-da44-4ae7-a0fc-84780a5f122c">

**Reply by nyamadan** [reply, marked as answer]
> UniVRM：0.111
> VRoid SDK：0.3.0
> VRoidマルチプレイSDK：0.3.0
> Unity：2020.3.43f1

こちらの環境で試してみましたが問題が再現しませんでした。

また、添付頂いたスクリーンショットにはSimpleURPToonLitOutlineExampleというシェーダが使用されていますが配布しているVRoid SDKにてSimpleURPToonLitOutlineExampleというシェーダを見つけることができませんでしたので、Exampleに何か別のスクリプトなどが追加されてしまっているのかもしれません。

また、やや古いUniVRMとUnityを使用していらっしゃるようですが Unity 2021及びUniVRM v0.113.0でも同様でしょうか？
